proposals/discard-parameters.md

```diff
@@ -0,0 +1,26 @@
+# Discard parameters
+
+## Summary
+
+Allow discards (`_`) to be used as parameters.
+For example:
+- lambdas: `(_, _) => 0`, `(int _, int _) => 0`
+- anonymous methods: `delegate(int _, int _) { return 0; }`
+- local functions: `void M() { void local(int _, int _) => ...; ... }`
+- methods: `override void M(int _, int _) { ... }`
+- indexers, constructors
+- delegate types: `delegate void MyDelegate(int _, int _);`
+
+## Motivation
+
+Unused parameters do not need to be named. The intent of discards is clear, i.e. they are unused/discarded.
+
+## Detailed design
+
+In a parameter list with more than one parameter named `_`, such parameters are discard parameters.
+Note: if a single parameter is named `_` then it is a regular parameter for backwards compatibility reasons.
+
+Discard parameters do not introduce any names to any scopes.
```

Comment:
>do not introduce any names to any scopes [](start = 19, length = 40)

What name lookup changes are proposed to implement this intent in the specification?

Reply:
Which section of the spec do you refer to for "name lookup"?
I think I'm proposing that we don't change name lookup, but rather we change what names are added to what scopes.
"Names are introduced into this declaration space by the type parameter list and the formal parameter list of the method and by local variable declarations in the block of the method." ([method parameters](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#method-parameters)) would be affected by the rule that discard parameters don't introduce a name in any scope.